x-pack/legacy/plugins/monitoring/public/np_imports/angular/index.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import angular, { IModule } from 'angular';
+
+import {
+  AppMountContext,
+  LegacyCoreStart
+} from 'kibana/public';
+
+import { configureAppAngularModule } from '../legacy_imports';
+
+//@ts-ignore TODO: change to absolute path
+import uiRoutes from 'plugins/monitoring/np_imports/ui/routes';
+//@ts-ignore TODO: change to absolute path
+import { uiModules } from 'plugins/monitoring/np_imports/ui/modules';
+//@ts-ignore TODO: change to absolute path
+import { registerTimefilterWithGlobalState } from 'plugins/monitoring/np_imports/ui/timefilter';
+
+import { localAppModule, appModuleName } from './modules';
+
+export class AngularApp {
+
+  private _injector?: angular.auto.IInjectorService;
+  private _observer?: MutationObserver;
+
+  constructor({ core }: AppMountContext, { appBasePath, element }: { appBasePath: string, element: HTMLElement }) {
+
+    const inDOM = () => document.body.contains(element);
```

Comment:
just curious why we need this?

Reply:
I was doing something asynchronous, and I think the root element was not yet attached to the DOM. It was causing some errors with `react-dom`  render() since the container(child) element was also not in DOM (in some cases we use document.getElement*...).